### PR TITLE
Improve developer instructions and replace the send-webhook script

### PR DESCRIPTION
**Pull request by rudymatela**
Closes: #125 which is a pre-requisite to #77.
Closes: #128 

This PR improves developer instructions with a step by step guide on how to run Hoff locally.  In addition, it adds a new `send-webhook` script with some common actions for local debugging.

Reviewer: please look at the relevant section on the [rendered readme.md file](https://github.com/channable/hoff/blob/doc/improve-readme-dev-instructions/readme.md#running-as-a-developer) as well as the diff.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -61,18 +61,63 @@ Hoff is written in Haskell and builds with [Stack][stack]:
...
+	$ ./tools/build-status c033170123456789abcdef0123456789abcdef01
+
+[Comment]: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#issue_comment
+[build status]: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#status
+
+The tests of Hoff are extensive, you may be able to get by just by running them
+when making changes to the code.  To run a specific test, you can temporarily
+add `(FOCUS)` to your test title and run the following:
+
+	$ nix run -c stack test --ta '--match "(FOCUS)"'
```
Comment (fatho) [inline readme.md:107]: Or just put (part of) the test title in the `--match` argument rather than `(FOCUS)`.

**Review by fatho — COMMENTED** [review]
Thanks for the detailed instructions, really nice!

I'd probably have kept `send-webhook` as a Python script (partially for reasons mentioned in the comments). With libraries such as `click` and `requests`, it should be pretty easy to put together.

Since it's just for testing, it's also fine as it is. But maybe for the next iteration :)

File: readme.md (hunk 1 above)
Comment (rudymatela) [reply readme.md:107]: I have updated this to refer to part of the test title instead.

File: tools/send-webhook [hunk 2]
```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (rudymatela) [reply tools/send-webhook:1]: I did now.  There were a couple of unquoted arguments which I fixed.  I also replaced the legacy back ticks with the `$()` notation.

File: tools/build-status [hunk 3]
```diff
@@ -0,0 +1 @@
+send-webhook
```
[the remark this answers is not in the document]
Comment (rudymatela) [reply tools/build-status:1]: Yes.  I have added a line in the preamble comment of send-webhook explaining this.

File: tools/send-webhook [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+# Takes the first occurrence of a field in config.json.
+# This is not a proper JSON parsing, but is enough
+# for the purposes of locally simulating webhooks
+# in a development environment.
+get() {
+	cat config.json |
+	grep $1 |
+	head -1 |
+	sed "s/.*\"$1\": *\"//;s/\".*//"
```
[the remark this answers is not in the document]
Comment (rudymatela) [reply tools/send-webhook:64]: It crossed my mind and would have been more robust.  But I'll leave it with a "grepsed" for now.

I've golfed the code a bit though now:

```bash
    grep -m1 "$1" config.json |
    sed "s/.*\"$1\": *\"//;s/\".*//"
```

**Comment by rudymatela** [comment]
Thanks for the review @fatho.  :smile:   I have amended the PR following your comments, though I still left `send-webhook` as a shell script for now.

@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Rebased as 5101b9df3f09b5d9b37a76fb4a5d6b063f82f37f, waiting for CI …

**Pull request merged**